src/dvclive/monitor_system.py

```diff
@@ -0,0 +1,195 @@
+        plot: bool = True,
+    ):
+        """Monitor CPU resources and log them to DVC Live.
+
+        Args:
+            interval (float): interval in seconds between two measurements.
+                Defaults to 0.5.
+            num_samples (int): number of samples to average. Defaults to 10.
+            disks_to_monitor (Optional[Dict[str, str]]): paths to the disks or
+                partitions to monitor disk usage statistics. The key is the name that
+                will be displayed in the metric name. The value is the path to the disk
+                or partition. Defaults to {"main": "/"}.
+            plot (bool): should the system metrics be saved as plots. Defaults to True.
+
+        Raises:
+            TypeError: if the arguments passed to the function don't have a
+                supported type.
+        """
+        super().__init__(interval=interval, num_samples=num_samples, plot=plot)
+        disks_to_monitor = (
+            {"main": "/"} if disks_to_monitor is None else disks_to_monitor
+        )
+        for disk_name, disk_path in disks_to_monitor.items():
+            if not isinstance(disk_name, str):
+                raise TypeError(  # noqa: TRY003
+                    "Keys for `partitions_to_monitor` should be a `str`"
+                    f", but got '{type(disk_name)}'"
+                )
+            if not isinstance(disk_path, str):
+                raise TypeError(  # noqa: TRY003
+                    "Value for `partitions_to_monitor` should be a `str`"
+                    f", but got '{type(disk_path)}'"
+                )
+            if disk_name != os.path.normpath(disk_name):
+                raise ValueError(  # noqa: TRY003
+                    "Keys for `partitions_to_monitor` should be a valid name"
+                    f", but got '{disk_name}'"
+                )
+        self._disks_to_monitor = disks_to_monitor
+
+    def _get_metrics(self) -> Dict[str, Union[float, int]]:
+        ram_info = psutil.virtual_memory()
+        nb_cpus = psutil.cpu_count()
+        cpus_percent = psutil.cpu_percent(percpu=True)
+        result = {
+            METRIC_CPU_COUNT: nb_cpus,
+            METRIC_CPU_USAGE_PERCENT: mean(cpus_percent),
+            METRIC_CPU_PARALLELIZATION_PERCENT: len(
+                [
+                    percent
+                    for percent in cpus_percent
+                    if percent >= MINIMUM_CPU_USAGE_TO_BE_ACTIVE
+                ]
+            )
+            * 100
+            / nb_cpus,
+            METRIC_RAM_USAGE_PERCENT: ram_info.percent,
+            METRIC_RAM_USAGE_GB: (ram_info.percent / 100)
+            * (ram_info.total / GIGABYTES_DIVIDER),
+            METRIC_RAM_TOTAL_GB: ram_info.total / GIGABYTES_DIVIDER,
+        }
+        for disk_name, disk_path in self._disks_to_monitor.items():
+            if not Path(disk_path).exists():
+                continue
+            disk_info = psutil.disk_usage(disk_path)
```

Comment:
I think it's better to catch OSError in this case instead of the `not Path` check (more reliable way)

Reply:
I follow the same pattern as the other parameters. If there is an OSError, the program continues, but we won't track the disk usage for the non-existing folder.